File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        vector<vector<int>> res;
        vector<int> current=nums;
        res.push_back(current);
        while(next_permutation(nums.begin(),nums.end())) {
            current = nums;
            res.push_back(current);
        }

        return res;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def isHappy(self, n: int) -> bool:
        s = set()
        while True:
            s.add(n)
            c=n
            n=0
            while c:
                n+=(c%10)**2
                c//=10
            if n==1: return True
            if n in s: return False
